### PR TITLE
Added support for SAML auth, IP routes, DNS Suffixes, Adding CA Stores, generating SAML Metadata.

### PR DESCRIPTION
**Pull request by TyphoidMary**
The module as provided in the blogs is fantastic but was missing some features for my use case, I added them and want to push them upstream for the next guy. The SAML stuff I think would be very helpful because there's not a lot of documentation for using the NetScaler as a SAML SP and this both goes through the config and generates the SAML2.0 Entity Metadata.

### CODE REVIEW

File: Modules/NetScalerConfiguration/NetScalerConfiguration.psm1 [hunk 1]
```diff
@@ -15,6 +15,26 @@ Set-StrictMode -Version Latest
 # Define default URL protocol to https, which can be changed by calling Set-Protocol function
 $Script:NSURLProtocol = "https"
 
+function Set-NSMgmtProtocol {
+    <#
+    .SYNOPSIS<#
```
Comment (santiagocardenas) [inline Modules/NetScalerConfiguration/NetScalerConfiguration.psm1]: Can you please fix this extra Synopsis?

File: Modules/NetScalerConfiguration/NetScalerConfiguration.psm1 [hunk 2]
```diff
@@ -15,6 +15,26 @@ Set-StrictMode -Version Latest
 # Define default URL protocol to https, which can be changed by calling Set-Protocol function
 $Script:NSURLProtocol = "https"
 
+function Set-NSMgmtProtocol {
```
Comment (santiagocardenas) [inline Modules/NetScalerConfiguration/NetScalerConfiguration.psm1]: Why two Set-NSMgmtProtocol functions?

**Review by santiagocardenas — CHANGES_REQUESTED** [review]
Nice PR! Can you please correct what seems to be duplicated code?

File: Modules/NetScalerConfiguration/NetScalerConfiguration.psm1 (hunk 2 above)
Comment (TyphoidMary) [reply Modules/NetScalerConfiguration/NetScalerConfiguration.psm1]: Removed dupe in 4a8bd7a

**Review by santiagocardenas — APPROVED** [review]

**Comment by santiagocardenas** [comment]
Thank you @TyphoidMary for contributing new functionality!

**Pull request merged**